lib/pbench/server/api/resources/upload_api.py

```diff
@@ -237,15 +231,12 @@ def put(self, controller):
         response.status_code = 201
         return response
 
+    ALLOWED_EXTENSION = ".tar.xz"
+
```

Comment:
It might make sense to have this defined "way down here" if it were only referenced in `allowed_file` ... but it's referenced up above for the error message, and "hiding" it down here makes it hard to find. It'd be better up at the top of the class.

Also, you reference it in `allowed_file` as `Upload.ALLOWED_EXTENSION` (which is clear and generally preferred) but as `self.ALLOWED_EXTENSION` above, implying an instance attribute rather than a class attribute. Both work, but referencing as a class attribute is better and they should at least be consistent.

Reply:
Moved to #2287.